src/core/coap/coap.cpp

```diff
@@ -829,14 +829,15 @@ uint32_t CoapTransmissionParameters::CalculateInitialRetransmissionTimeout(void)
 uint32_t CoapTransmissionParameters::CalculateExchangeLifetime(void) const
 {
     uint32_t maxTransmitSpan =
-        mAckTimeout * ((2 << (mMaxRetransmit - 1)) - 1) * mAckRandomFactorNumerator / mAckRandomFactorDenominator;
+        mAckTimeout * (((uint64_t)1 << mMaxRetransmit) - 1) * mAckRandomFactorNumerator / mAckRandomFactorDenominator;
     uint32_t processingDelay = mAckTimeout;
     return maxTransmitSpan + 2 * kDefaultMaxLatency + processingDelay;
 }
 
 uint32_t CoapTransmissionParameters::CalculateMaxTransmitWait(void) const
 {
-    return mAckTimeout * ((2 << mMaxRetransmit) - 1) * mAckRandomFactorNumerator / mAckRandomFactorDenominator;
+    return mAckTimeout * (((uint64_t)2 << mMaxRetransmit) - 1) * mAckRandomFactorNumerator /
```

Comment:
I think `static_cast<uint64_t>(expr)` is normally preferred in C++ files… but given this is a constant, not sure what the policy is.

Reply:
Changed